packages/headless/src/controllers/search-box/headless-search-box-options.ts

```diff
@@ -35,16 +35,24 @@ export interface SearchBoxOptions {
    * @defaultValue `5`
    */
   numberOfSuggestions?: number;
+
+  /**
+   * Specifies whether to clear filters when the user enters a new search query
+   *
+   * @defaultValue 'false'
```

Comment:
Aside from trying to avoid a change in behaviour, is there a use-case for having this option default to `false`?

When I look at google, youtube and amazon, they all clear filters on a new search.

So I'm wondering if this is a bug in the headless searchbox controller's UX, or is there a use-case for wanting to toggle the clear filter behaviour? Said differently, have you seen any sites that keep filters when performing a search?

Reply:
It would be a breaking behaviour change I guess? If we don't see it as issue, let's set it to true by default like for Atomic

Reply:
I mean... I'm fine if we want to "force" the clear on new query. I won't die on this hill.

However, I do find myself doing precisely what you describe @samisayegh , whenever I search on our docsite, or Coveo@Coveo, things like that.

Where I submit a query, refine with a facet, then go back to the searchbox and further "refine", again, by adding more keywords in the searchbox, and hopefully by then I'm out of the search interface because I found the document I was looking for.

It's actually rare that I do a search, use a facet, then decide to actually search for something completely unrelated in the "same session" (aka: the same page load lifecycle).

However, there are some other time, like for example commerce, where I might be using the search interface not as a way to "find a document", but instead to "browse", where my initial search intention will change over time. And so here, I'd be nice if the filters were cleared automatically for me.

So 🤷 ...

But as I said, I'm fine with not having any option.

It is however a change in behaviour, that could be considered breaking.

Reply:
I understand sometimes you might want to refine a query after having selected filters, but I think it's important to consider the other side, since the option is either `true` or `false`. The UX will apply to all queries; there isn't an easy way to detect the intent behind the search.

The sites I looked at serve different use-cases: e-commerce (Amazon, Ebay, Dell), general search (Google), video search (Youtube). All clear filters on search.

I read through links JF shared [here](https://discuss.coveo.com/t/clear-facet-selection-for-a-new-query-or-not/2347) and [here](https://coveord.atlassian.net/wiki/spaces/UX/pages/723124387/Clearing+Filters+on+New+Query). The UX recommendation is towards clearing filters.

If it was anything other than 100 - 0 clear/do not clear, I would consider adding a clearingFilters option, but atm I see it as option creep that leads to worse UX on average. A customer who absolutely does not want to use the recommended UX can still create the behavior by dispatching actions.

A behavior change can be considered breaking, but in this case I see it as a bug-fix with headless's state management to impove UX.